Allow failures on CI changelog reminder

mschwager opened the pull request:
https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error

The changelog reminder keeps failure our CI runs. This isn't a crucial task that should fail runs, in my opinion.

Commit:
Allow failures on CI changelog reminder

underyx commented:
Can you link an example changelog? It was not supposed to ever fail. This might signal a crash or other misconfiguration that would be better handled with a fix to the job.

underyx reviewed (changes requested):
Can you link an example changelog failure? It was not supposed to ever fail. This might signal a crash or other misconfiguration that would be better handled with a fix to the job.

underyx commented:
> Can you link an example changelog failure? It was not supposed to ever fail. This might signal a crash or other misconfiguration that would be better handled with a fix to the job.

I found an example failure and I see it happens when the job runs on develop since https://github.com/returntocorp/semgrep/pull/815 set it up like that. It was not meant to, as the job doesn't make sense on non-PR jobs.

The correct fix is to either move it back to its own workflow where it only runs on PRs, or to somehow conditionally skip this one job of the workflow on push events.

Also, I haven't been very active in this repo since I added the job. Has it actually proven to be useful? If not, might as well just remove it instead. What do you think @brendongo @mschwager?

Commit:
Only run changelog reminder on PRs

mschwager commented:
> The correct fix is to either move it back to its own workflow where it only runs on PRs, or to somehow conditionally skip this one job of the workflow on push events.

I've updated the workflow to only run the changelog reminder on PRs. Does that work?

> Also, I haven't been very active in this repo since I added the job. Has it actually proven to be useful? If not, might as well just remove it instead. What do you think @brendongo @mschwager?

It's been somewhat useful for me. There's been a few times where it has in fact reminded me to bump the CHANGELOG, but I'd say it's unnecessarily added the message about 90% of the time.

underyx commented:
Let's leave the keep or remove decision up to @brendongo then. I can't quite reason about the value of 1 true positive vs. cost of 9 false positives when it's just a non-blocking info comment.
